### PR TITLE
Filebeat 5.0.0

### PR DESCRIPTION
**Pull request by ruflin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is in line with the metricbeat 5.0.0 Formula https://github.com/Homebrew/homebrew-core/pull/6441

### CODE REVIEW

File: Formula/filebeat.rb [hunk 1]
```diff
@@ -1,19 +1,11 @@
...
+  desc "File harvester to ship log files to Elasticsearch or Logstash"
   homepage "https://www.elastic.co/products/beats/filebeat"
-  url "https://github.com/elastic/beats/archive/v1.2.3.tar.gz"
-  sha256 "8eea85de415898c362144ba533062651d8891241c738799e54cc9b17040c1fc9"
+  url "https://github.com/elastic/beats/archive/v5.0.0.tar.gz"
+  sha256 "3e6b7cf2ee5f52e78ae87ef04ab9dd49977c89f86a09416586896aeaea844e34"
 
   head "https://github.com/elastic/beats.git"
 
-  bottle do
```
Comment (ruflin) [inline Formula/filebeat.rb]: Ok, thought it had to be removed every time as it is auto generated. Still new to homebrew contributions. Put it back in.

**Comment by ruflin** [comment]
@ilovezfs Thanks for the review. I updated the PR according to your comments.

File: Formula/filebeat.rb [hunk 2]
```diff
@@ -25,12 +17,15 @@ def install
...
+
+      (etc/"filebeat").install "filebeat.yml"
+      (etc/"filebeat").install "filebeat.template.json"
+      (etc/"filebeat").install "filebeat.template-es2x.json"
     end
 
     (bin/"filebeat").write <<-EOS.undent
       #!/bin/sh
-      exec "#{libexec}/filebeat" -c "#{etc}/filebeat.yml" "$@"
+      exec "#{libexec}/filebeat" --path.config "#{etc}/filebeat" --path.home="#{prefix}" --path.logs="#{var}/log/filebeat" --path.data="#{opt_prefix}" "$@"
```
Comment (ilovezfs) [inline Formula/filebeat.rb]: The files in /usr/local/opt are symlinks that change every time the installation is upgraded to point to the current Cellar path. So the path is stable, but the directory it points to is changing every time.

Anything that persists across versions probably belongs in one of
```
etc
etc/"foo"
var
etc/"foo"
```
etc and var are different from bin, lib, include, share, ... in that they are locations in the `HOMEBREW_PREFIX` not the formula `prefix`. So `etc` is the directory `/usr/local/etc` for all formulae and `var` is the directory `/usr/local/var` for all formulae.

Comment (ruflin) [inline Formula/filebeat.rb]: Based on your input I changed it to `--path.data="#{var}/filebeat"`.

**Review by ruflin — COMMENTED** [review]
New version with the requested changes pushed.

**Comment by MikeMcQuaid** [comment]
Test is failing trying to write into prefix: https://bot.brew.sh/job/Homebrew%20Core/10867/version=any_sierra/testReport/junit/brew-test-bot/sierra/test_filebeat/

**Comment by ruflin** [comment]
Is there something special I need to do here or is it just a bad idea to create a directory in var? Will it help if I let the Formula create the directory if it does not exist yet?

**Comment by MikeMcQuaid** [comment]
The `test` cannot write to `/usr/local/var`.

**Comment by ruflin** [comment]
@MikeMcQuaid Makes sense. I adjusted the tests so all data goes to the temporary testing directory.

**Comment by MikeMcQuaid** [comment]
Still seems to be trying to write to `var`: https://bot.brew.sh/job/Homebrew%20Core/10988/version=any_sierra/testReport/junit/brew-test-bot/sierra/test_filebeat/

**Comment by ruflin** [comment]
@MikeMcQuaid Sorry about that. I think I finally got it right.

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**